clients/tabby-agent/src/codeCompletion/contexts.ts

```diff
@@ -132,9 +156,50 @@ export class CompletionContext {
       git: this.git,
       declarations: this.declarations,
       relevantSnippetsFromChangedFiles: this.relevantSnippetsFromChangedFiles,
+      relevantSnippetsFromOpenedFiles: this.snippetsFromOpenedFiles,
+      autoComplete: {
+        completionItem: this.completionItem,
+        insertPosition: this.insertPosition,
+        insertSeg: this.insertSeg,
+        currSeg: this.currSeg,
+      },
```

Comment:
I believe the forward cache generating will be broken for the case with `selectedCompletionInfo`. We can fix it in next PRs.
```suggestion
      // FIXME: generate the context according to `selectedCompletionInfo`
```